app-modules/student-data-model/src/Actions/CreateTemporaryStudentDataImportTables.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2025, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\StudentDataModel\Actions;
+
+use App\Models\Import;
+use Illuminate\Support\Facades\DB;
+
+class CreateTemporaryStudentDataImportTables
+{
+    public function execute(
+        ?Import $studentsImport,
+        ?Import $programsImport = null,
+        ?Import $enrollmentsImport = null,
+    ): void {
+        DB::transaction(function () use ($studentsImport, $programsImport, $enrollmentsImport) {
+            DB::statement("create table \"import_{$studentsImport->getKey()}_students\" as table \"students\" with no data");
+
+            if ($programsImport) {
+                DB::statement("create table \"import_{$programsImport->getKey()}_programs\" as table \"programs\" with no data");
+            }
+
+            if ($enrollmentsImport) {
+                DB::statement("create table \"import_{$enrollmentsImport->getKey()}_enrollments\" as table \"enrollments\" with no data");
+            }
```

Comment:
Hmm, is there a particular reason you chose `as table` over `CREATE ... LIKE`? (https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-PARMS-LIKE)

LIKE has the ability to copy constraints, indexes, etc. while `as table` does not.

Reply:
Done